activiti-cloud-api-process-model-impl/src/main/java/org/activiti/cloud/api/process/model/impl/events/CloudSequenceFlowTakenImpl.java

```diff
@@ -28,7 +28,8 @@ public CloudSequenceFlowTakenImpl() {
 
     public CloudSequenceFlowTakenImpl(SequenceFlow sequenceFlow) {
         super(sequenceFlow);
-        setEntityId(sequenceFlow.getProcessInstanceId());
+        if (getEntity()!=null)
+            setEntityId(getEntity().getSourceActivityElementId());
```

Comment:
Here, you also need to use `getEntity.getId()`after fixing https://github.com/Activiti/Activiti/issues/2175.

Reply:
1. Do we need this for BPMNElement (as it is stated in Issue) or BPMNActivity?
public interface BPMNActivity extends BPMNElement
2. setEntityId(getEntity().getSourceActivityElementId());
For SequenceFlow we have SourceActivity, and Target...
O rI really have to write: setEntityId(getEntity().getEntity.getId());

Reply:
@CTI777 this is related to the closed PR in Activiti-api, here you should use SequenceFlow.getElementId() in the same way as the BPMNActivity after changing that in Activiti API